### PR TITLE
feat: rewrite Tooltip using popover api

### LINKED ISSUE #249: Rewrite Tooltip using Popover api
Works nicely, but there is quite a significant amount of code to it. I wonder if this could be made simpler by using the [Popover API](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API) which is well supported today.

_Originally posted by @agazso in https://github.com/diete-design/diete.design/pull/191#pullrequestreview-2098962022_

### PR DESCRIPTION
**Pull request by hlavkat**
close #249

### CODE REVIEW

**Review by agazso — CHANGES_REQUESTED** [review]
Does not work correctly on Firefox.

Also the info icons don't appear in the correct color in dark mode.
Screenshot_20240613_114214-1

File: src/lib/components/ui/tooltip.svelte [hunk 1]
```diff
@@ -1,119 +1,157 @@
...
-		}
-	}
+	let {
+		position = 'top',
+		children,
+		helperText,
+		large = false,
+		show = false,
+		tooltipId = Math.random().toString(16),
+		tooltipName = 'tooltip',
```
Comment (agazso) [inline src/lib/components/ui/tooltip.svelte:20]: Can this be randomised too?
```suggestion
		tooltipName = 'tooltip' + Math.random().toString(16),
```

**Comment by vojtechsimetka** [comment]
Did I understood it correctly, that we ice this for now since the popover API is missing some crucial features in Mozilla?

**Comment by agazso** [comment]
> Did I understood it correctly, that we ice this for now since the popover API is missing some crucial features in Mozilla?

Yes, that is my understanding. @hlavkat it would be good to document here what is the issue and close this PR.

**Comment by hlavkat** [comment]
The Popover API has quite complex positioning that doesn't work in Firefox, it's a relatively new attribute and is still being developed, so we are waiting for upcoming improvements in positioning and for the popover attribute to mature further.

**Pull request closed without merging**